src/ezsp/ezsp-dongle.cpp

```diff
@@ -215,3 +239,24 @@ void CEzspDongle::notifyObserversOfEzspRxMessage( EEzspCmd i_cmd, std::vector<ui
 		observer->handleEzspRxMessage(i_cmd, i_message);
 	}
 }
+
+void CEzspDongle::handleDongleState( EDongleState i_state )
+{
+	// do nothing
+}
+
+void CEzspDongle::handleEzspRxMessage( EEzspCmd i_cmd, std::vector<uint8_t> i_msg_receive )
+{
+    // response to a sending command
+	if( wait_rsp && !sendingMsgQueue.empty() )
+	{
+		sMsg l_msgQ = sendingMsgQueue.front();
+		if( l_msgQ.i_cmd == i_cmd ) // Bug
+		{
+			// remove waiting message and send next
+			sendingMsgQueue.pop();
+			wait_rsp = false;
+			sendNextMsg();
+		}
+	}
+}
```

Comment:
I read and read again, and it's smell good for me. But I'm not able to test a lot.
Give me your point of view. @sebastienmarion ?

Reply:
I think both are actually equivalent. My note was more on the semantic that explicitly shows (in the same bloc) that we are dequeuing the next message after having received a response, and also because it is maybe better to notify all external observers before we send the next message in the queue (if I remember well, not specific order is guaranteed when notifying observers, so the call to CEzspDongle::handleEzspRxMessage() can be done before, after, or in the middle of other observers notification).